### PR TITLE
699 - new header

### PR DESCRIPTION
**Pull request by jrgarou**
#699 

image

### CODE REVIEW

File: src/scripts/components/foundations/Button.js [hunk 1]
```diff
@@ -40,16 +69,38 @@ export const StyleAnchor = css`
...
+    props.anchor
+      ? props.theme.fonts.weight.regular
+      : props.theme.fonts.weight.bold};
+  margin: ${props => props.margin};
+  opacity: ${props => (props.disabled ? 0.5 : null)};
+  pointer-events: ${props => (props.disabled ? 'none' : null)};
+  text-transform: ${props => (props.anchor ? null : 'uppercase')};
+  user-select: ${props => (props.anchor ? null : 'none')};
+  text-rendering: geometricPrecision;
+`
```
Comment (bent0b0x) [inline src/scripts/components/foundations/Button.js]: Can we move these to another file? Is this a good PR to refactor this component to be more generic? Fine if not 😄

File: src/scripts/components/foundations/Button.js [hunk 2]
```diff
@@ -40,16 +69,38 @@ export const StyleAnchor = css`
...
+    filter: brightness(1.3);
   }
 `
 
+export const ButtonColor = styled.button`
+  ${ButtonStyle} ${ButtonStyleHover} ${ButtonStyleBackground} border-radius: 2px;
+  color: ${props => props.theme.colors.button.white};
+  padding: 8px 18px;
+`
+
```
Comment (bent0b0x) [inline src/scripts/components/foundations/Button.js]: Also, in general I think we may want to examine this component a bit. I tried to use it for tipping but found some bugs w/ the active/hover states that I could not easily work around. Essentially the button appeared in those states inappropriately after clicking on and off. 

Just an FYI (have you seen issues like this?)

File: src/scripts/components/foundations/forms/Input.js [hunk 3]
```diff
@@ -9,10 +9,16 @@ const AutofillFix = props => keyframes`
...
+  color: ${props => props.theme.colors.TextField.color};
   font-family: ${props => props.theme.fonts.family}, sans-serif;
   font-size: ${props => props.theme.fonts.form.input};
   font-weight: ${props => props.theme.fonts.weight.regular};
   width: 100%;
+
+  &:-webkit-autofill {
+    animation-name: ${AutofillFix};
+    animation-fill-mode: both;
+  }
```
Comment (bent0b0x) [inline src/scripts/components/foundations/forms/Input.js]: What about other browsers?

Comment (bent0b0x) [reply src/scripts/components/foundations/forms/Input.js]: Ah I see this was just moved. But what about other browsers?

File: src/scripts/components/structures/header/MainNavigation.js [hunk 4]
```diff
@@ -106,7 +98,7 @@ class MainNavigation extends Component<Props, Object> {
 
           {!this.props.isWalletSecured ? (
             <NavItem>
-              <NavLinkPurple
+              <ButtonColor
```
Comment (bent0b0x) [inline src/scripts/components/structures/header/MainNavigation.js]: The name `ButtonColor` seems confusing. That should be a color variable, I would think. What about `<ColoredButton />`?

File: src/scripts/components/structures/header/MainNavigation.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ import React, { Component } from 'react'
 import { Link } from 'react-router-dom'
 import styled from 'styled-components'
 
-import Button from 'components/foundations/Button'
+import Button, { ButtonColor } from 'components/foundations/Button'
```
Comment (bent0b0x) [inline src/scripts/components/structures/header/MainNavigation.js]: Can we move `ButtonColor` to another file?

**Comment by jellegerbrandy** [comment]
ok @jrgarou , can you address ben's comments? Then feel free to merge :-)

File: src/scripts/components/foundations/Button.js (hunk 1 above)
Comment (jrgarou) [reply src/scripts/components/foundations/Button.js]: @bent0b0x let's create an issue for all those changes? 
What I think about: 
- Button(colored button)
- TextButton(actually Button)

And share(`export`) its styles.

File: src/scripts/components/foundations/Button.js (hunk 2 above)
Comment (jrgarou) [reply src/scripts/components/foundations/Button.js]: >have you seen issues like this?

Probably some `:active` selector. Could you send more info about this, please?

File: src/scripts/components/foundations/forms/Input.js (hunk 3 above)
Comment (jrgarou) [reply src/scripts/components/foundations/forms/Input.js]: this is only for the yellow background from webkit

File: src/scripts/components/structures/header/MainNavigation.js (hunk 4 above)
Comment (jrgarou) [reply src/scripts/components/structures/header/MainNavigation.js]: Could we do those changes with **[this](https://github.com/Paratii-Video/paratii-portal/pull/730#discussion_r196427367)** ?

File: src/scripts/components/structures/header/MainNavigation.js (hunk 5 above)
Comment (jrgarou) [reply src/scripts/components/structures/header/MainNavigation.js]: Could we do those changes with **[this](https://github.com/Paratii-Video/paratii-portal/pull/730#discussion_r196427367)** ?

File: src/scripts/components/foundations/Button.js (hunk 1 above)
Comment (bent0b0x) [reply src/scripts/components/foundations/Button.js]: I think we should definitely create an issue to discuss further. 

Not sure about that distinction, though, and my initial thought is that we should export any styles from components. I believe there are other ways that we can share styles as needed, via constants and such. Component files should only be components, and should not have any other responsibilities like maintaining styles for other components (unless a component is a direct extension of said component). 

I also feel like if we need multiple components to share many of the same styles then perhaps we have not written our abstractions correctly, and we can refactor a bit

Comment (bent0b0x) [reply src/scripts/components/foundations/Button.js]: https://github.com/Paratii-Video/paratii-portal/issues/737

**Pull request merged**